src/Microsoft.OpenApi.OData.Reader/Common/OpenApiOperationExtensions.cs

```diff
@@ -36,7 +36,8 @@ public static void AddErrorResponses(this OpenApiOperation operation, OpenApiCon
 
         if(addNoContent)
 		{
-			operation.Responses.Add(Constants.StatusCode204, Constants.StatusCode204.GetResponse());
+            var statusCode = settings.UseHTTPStatusCodeClass2XX ? Constants.StatusCodeClass2XX : Constants.StatusCode204;
+            operation.Responses.Add(statusCode, Constants.StatusCode204.GetResponse());
```

Comment:
Won't this result in an empty (no content) payload for success status code range - 2xx? i.e., 
``` yml
responses:
  '204':
    description: Success
  default:
    $ref: '#/components/responses/error'
```

Clients will handle each success status code accordingly as discussed in https://github.com/Azure/autorest.powershell/issues/853#issuecomment-948617364.

We should always return schema in the response when `UseHTTPStatusCodeClass2XX` is `true`.
``` yml
responses:
  2XX:
    description: success
    content:
      application/json:
        schema: 
          type: object
```

Reply:
We still do:

![image](https://user-images.githubusercontent.com/40403681/165771877-a1bb32ed-dda6-4838-98f6-4fd87415ca7b.png)
_with_ `UseHTTPStatusCodeClass2XX = true`


![image](https://user-images.githubusercontent.com/40403681/165772098-297b6a69-857d-46d6-bc65-39efea1ec5e7.png)
_with_ `UseHTTPStatusCodeClass2XX = false`


Nothing in the initial response generation has been modified, just the response status code only (if `UseHTTPStatusCodeClass2XX = true`), everything else remains the same.

Reply:
I have added `Content` objects to the responses of POST, PATCH, PUT and UPDATE operations.